File: src/service/ModelBuilder.Models/Node.cs
using System;
using System.Collections.Generic;
using Mb.Models.Enums;

namespace Mb.Models
{
    public class Node
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string SemanticId { get; set; }
        public string TagNumber { get; set; }
        public string Name { get; set; }
        public IconType Icon { get; set; }
        public string Label { get; set; }
        public NodeType Type { get; set; }
        public decimal PositionX { get; set; }
        public decimal PositionY { get; set; }
        public virtual ICollection<Connector> Connectors { get; set; }
        public bool IsLocked { get; set; }
        public bool IsSelected { get; set; }

        public decimal PositionBlockX { get; set; }
        public decimal PositionBlockY { get; set; }
        public int Length { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }

        public string UpdatedBy { get; set; }
        public DateTime Updated { get; set; }


        public virtual ICollection<Attribute> Attributes { get; set; }
        public virtual ICollection<Project> Projects { get; set; }

        
    }
}

File: src/client/src/models/classes/Node.ts
import { EnumBase, Attribute, Aspect, Purpose, Composite } from "..";
import Connector from "./Connector";

export const NODE_KIND: string = "Node";

class Node {
  id: string;
  iri: string;
  domain: string;
  projectId: string;
  rds: string;
  description: string;
  semanticReference: string;
  tagNumber: string;
  name: string;
  label: string;
  positionX: number;
  positionY: number;
  purpose: Purpose;

  positionBlockX: number;
  positionBlockY: number;
  level: number;
  order: number;
  statusId: string;
  status: EnumBase;
  createdBy: string;
  libraryTypeId: string;
  created: Date;
  updatedBy: string;
  updated: Date;
  version: string;
  aspect: Aspect;
  isRoot: boolean | false;
  masterProjectId: string;
  masterProjectIri: string;
  symbol: string;
  connectors: Connector[];
  attributes: Attribute[];
  simples: Composite[];
  width: number;
  height: number;

  // Required only for product aspect
  cost: number;

  // Only for client
  isSelected: boolean | false;
  isBlockSelected: boolean | false;
  isHidden: boolean | false;
  blockWidth: number;
  blockHeight: number;

  isLocked: boolean;
  isLockedStatusBy: string;
  isLockedStatusDate: string;

  kind: string = NODE_KIND;

  constructor(node: Node) {
    Object.assign(this, node);
  }
}

export default Node;
